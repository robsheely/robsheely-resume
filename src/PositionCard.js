import React from "react";
import PropTypes from "prop-types";

import Card from "@material-ui/core/Card";
import CardContent from "@material-ui/core/CardContent";
import List from "@material-ui/core/List";
import ListItem from "@material-ui/core/ListItem";
import ListItemText from "@material-ui/core/ListItemText";
import Typography from "@material-ui/core/Typography";
import { withStyles } from "@material-ui/core/styles";

const styles = theme => ({
  root: {
    width: "100%",
    marginTop: 6,
    marginBottom: 6,
    paddingLeft: 20,
    paddingRight: 20
  },
  content: {
    textAlign: "left"
  },
  bullet: {
    display: "inline-block",
    margin: "0 2px"
  },
  item: {
    paddingTop: 0,
    paddingBottom: 8
  },
  title: {
    fontSize: 16,
    display: "flex",
    justifyContent: "space-between"
  }
});

const PositionCard = props => {
  const { classes, position } = props;
  const { role, start, end, company, achievements } = position;
  const companyId = company.replace(/\s+/g, "-").toLowerCase();

  return (
    <Card
      id={`position-${companyId}`}
      data-testid={`position-${companyId}`}
      className={classes.root}
    >
      <CardContent classes={{ root: classes.content }}>
        <Typography
          id={`position-${companyId}-title`}
          data-testid={`position-${companyId}-title`}
          className={classes.title}
        >
          <span>{role.toUpperCase()}</span>
          <span>{`${start} - ${end}`}</span>
        </Typography>
        <Typography
          id={`position-${companyId}-company`}
          data-testid={`position-${companyId}-company`}
          className={classes.title}
        >
          {company}
        </Typography>
        <List
          id={`position-${companyId}-achievements`}
          data-testid={`position-${companyId}-achievements`}
          dense={false}
        >
          {achievements.map(achievement => (
            <ListItem
              id={`position-${companyId}-achievement-${achievement.id}`}
              data-testid={`position-${companyId}-achievement-${achievement.id}`}
              className={classes.item}
              key={achievement.id}
            >
              <span className={classes.bullet}>â€¢</span>
              <ListItemText primary={achievement.content}/>
            </ListItem>
          ))
          }
        </List>
      </CardContent>
    </Card>
  );
};

PositionCard.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(PositionCard);
