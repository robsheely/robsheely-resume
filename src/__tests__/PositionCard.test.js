import React from "react";
import { render, cleanup } from "react-testing-library";

import PositionCard from "../PositionCard";

const MOCK_DATA = {
  position: {
    role: "Role",
    start: 2002, 
    end: 2012, 
    company: "Good Company", 
    achievements: [
      {
        id: "1",
        content: "Achievement 1"
      },
      {
        id: "2",
        content: "Achievement 2"
      },
      {
        id: "3",
        content: "Achievement 3"
      }
    ]
  }
};

describe("PositionCard", () => {
  afterEach(cleanup);

  it("Passes smoke test", () => {
    const { container } = render(<PositionCard position={MOCK_DATA.position}/>);

    expect(container).toMatchSnapshot();
  });

  it("Contains expected children & content", () => {
    const { queryByTestId } = render(<PositionCard position={MOCK_DATA.position}/>);

    expect(queryByTestId("position-good-company")).not.toBeNull();
    expect(queryByTestId("position-good-company-title").textContent).toBe("ROLE2002 - 2012");
    expect(queryByTestId("position-good-company-company").textContent).toBe("Good Company");
    expect(queryByTestId("position-good-company-achievements")).not.toBeNull();
    expect(queryByTestId("position-good-company-achievements").children.length).toBe(3);
    expect(queryByTestId("position-good-company-achievement-2").textContent).toBe("â€¢Achievement 2");
  });
});