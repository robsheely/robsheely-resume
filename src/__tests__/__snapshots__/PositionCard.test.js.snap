// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionCard Passes smoke test 1`] = `
<div>
  <div
    class="MuiPaper-root-7 MuiPaper-elevation1-10 MuiPaper-rounded-8 MuiCard-root-6 PositionCard-root-1"
    data-testid="position-good-company"
    id="position-good-company"
  >
    <div
      class="MuiCardContent-root-34 PositionCard-content-2"
    >
      <p
        class="MuiTypography-root-35 MuiTypography-body1-44 PositionCard-title-5"
        data-testid="position-good-company-title"
        id="position-good-company-title"
      >
        <span>
          ROLE
        </span>
        <span>
          2002 - 2012
        </span>
      </p>
      <p
        class="MuiTypography-root-35 MuiTypography-body1-44 PositionCard-title-5"
        data-testid="position-good-company-company"
        id="position-good-company-company"
      >
        Good Company
      </p>
      <ul
        class="MuiList-root-71 MuiList-padding-72"
        data-testid="position-good-company-achievements"
        id="position-good-company-achievements"
      >
        <li
          class="MuiListItem-root-75 MuiListItem-default-78 MuiListItem-gutters-83 PositionCard-item-4"
          data-testid="position-good-company-achievement-1"
          id="position-good-company-achievement-1"
        >
          <span
            class="PositionCard-bullet-3"
          >
            •
          </span>
          <div
            class="MuiListItemText-root-87"
          >
            <span
              class="MuiTypography-root-35 MuiTypography-subheading-42 MuiListItemText-primary-90"
            >
              Achievement 1
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root-75 MuiListItem-default-78 MuiListItem-gutters-83 PositionCard-item-4"
          data-testid="position-good-company-achievement-2"
          id="position-good-company-achievement-2"
        >
          <span
            class="PositionCard-bullet-3"
          >
            •
          </span>
          <div
            class="MuiListItemText-root-87"
          >
            <span
              class="MuiTypography-root-35 MuiTypography-subheading-42 MuiListItemText-primary-90"
            >
              Achievement 2
            </span>
          </div>
        </li>
        <li
          class="MuiListItem-root-75 MuiListItem-default-78 MuiListItem-gutters-83 PositionCard-item-4"
          data-testid="position-good-company-achievement-3"
          id="position-good-company-achievement-3"
        >
          <span
            class="PositionCard-bullet-3"
          >
            •
          </span>
          <div
            class="MuiListItemText-root-87"
          >
            <span
              class="MuiTypography-root-35 MuiTypography-subheading-42 MuiListItemText-primary-90"
            >
              Achievement 3
            </span>
          </div>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
